backend/store/principal.go

```diff
@@ -430,12 +440,30 @@ func (s *Store) UpdateUser(ctx context.Context, userID int, patch *UpdateUserMes
 		&user.Name,
 		&user.Type,
 		&user.PasswordHash,
+		&idpID,
+		&idpUserInfo,
 	); err != nil {
 		if err == sql.ErrNoRows {
 			return nil, nil
 		}
 		return nil, FormatError(err)
 	}
+	if idpID.Valid {
+		value := int(idpID.Int32)
+		idp, err := s.GetIdentityProvider(ctx, &FindIdentityProviderMessage{
+			UID: &value,
+		})
+		if err != nil {
+			return nil, FormatError(err)
+		}
+		user.IdentityProviderResourceID = &idp.ResourceID
```

Comment:
Do we actually need to get the idp here at all? 🤔 we are only updating the `idp_user_info` column of the `principal` table but nothing from the `idp` table?